tests/base/src/test/java/org/wildfly/security/sasl/test/SaslServerBuilder.java

```diff
@@ -290,6 +294,11 @@ public SaslServerBuilder registerSecurityDomainReference(BuilderReference<Securi
         return this;
     }
 
+    public SaslServerBuilder setProviderSupplier(Supplier<Provider[]> providerSupplier) {
+        this.providerSupplier = providerSupplier;
```

Comment:
It would good to have an `Assert.assertNotNull` check here to make sure the `providerSupplier` being passed in isn't null.